generators/client/templates/react/src/main/webapp/app/shared/util/pagination-component.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';
+
+export interface IPaginationComponentProps {
+  items: number;
+  activePage: number;
+  onSelect: any;
+  maxButtons: number;
+}
+
+export interface IPaginationComponentState {
+  currentPage: number;
+}
+
+export class PaginationComponent extends React.Component<IPaginationComponentProps, IPaginationComponentState> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentPage: this.props.activePage
+    };
+  }
+
+  updateActivePage = currentPage => () => {
+    this.setState({ currentPage });
+    this.props.onSelect(currentPage);
+  }
+
+  previousPage = () => {
+    this.setState({ currentPage: this.state.currentPage - 1 });
+    this.props.onSelect(this.state.currentPage - 1);
+  }
+
+  nextPage = () => {
+    this.setState({ currentPage: this.state.currentPage + 1 });
+    this.props.onSelect(this.state.currentPage + 1);
+  }
+
+  itemsToDisplay = activePage => {
+    const items = [];
+    let item: any = {};
```

Comment:
`: any` is not needed as its inferred from value

Reply:
I think it's mandatory if not I have this error
`TS2339: Property 'display' does not exist on type '{}'.`

Reply:
oh, ok. Strange though